content/300-guides/050-database/100-developing-with-prisma-migrate/100-advanced-migrate-scenarios.mdx

```diff
@@ -94,6 +94,61 @@ To rename the `biograpy` field to `biography`:
 
 You can use the same technique to rename a `model` - edit the generated SQL to _rename_ the table rather than drop and re-create it.
 
+## Using the expand and contract pattern to evolve the schema without downtime
+
+One of the challenges when renaming fields is that it can lead to downtime in the time frame between applying the rename migration and deloying a new version of the application code which uses the renamed field.
```

Comment:
Not quite as familiar with the context of this section so ignore if it's out of scope.  Just wanted to mention that the pattern can be used for more than renaming fields and if this section is supposed to be about a specific way to do field renaming, maybe this should be a subheading instead of a peer heading to the one above it.

Reply:
The goal was to make it as general as possible. I used the renaming example because it's one of the easiest to explain. 

Perhaps I should add a note to clarify this. 

Would be great if you can share some good examples of when to use the expand and contract pattern.

Reply:
Gotcha! That makes complete sense.  Yeah, just a quick heads up that the renaming is being used as an exemplar would sort that out I think.

In terms of other times when you can use the expand and contract pattern, there are lots of potential ways.  I use an example of combining two columns in my article on migration strategies: https://www.prisma.io/dataguide/types/relational/migration-strategies#example-using-expand-and-contract-and-feature-flags

The example towards the end of this article is a pretty good one for adding a mapping table between two tables that were previously directly connected: https://www.tim-wellhausen.de/papers/ExpandAndContract/ExpandAndContract.html

Reply:
> The example towards the end of this article is a pretty good one for adding a mapping table between two tables that were previously directly connected: https://www.tim-wellhausen.de/papers/ExpandAndContract/ExpandAndContract.html

That's a nifty example! Thanks for sharing.